File: frontend/models/order.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type OrderItem struct {
	ProductID   primitive.ObjectID `bson:"productID"`
	ProductName string             `bson:"productName"` // <-- Nuevo campo
	Quantity    int                `bson:"quantity"`
}
type Order struct {
	ID           primitive.ObjectID `bson:"_id,omitempty"`
	CustomerName string             `bson:"customerName"`
	City         string             `bson:"city"`
	Phone        string             `bson:"phone"`
	Items        []OrderItem        `bson:"items"`
	CreatedAt    primitive.DateTime `bson:"createdAt"`
}

File: backend/models/order.js
// /models/order.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const OrderSchema = new Schema({
  customerName: { type: String, required: true },
  city: { type: String, required: true },
  phone: { type: String, required: true },
  items: [
    {
      productId: { type: Schema.Types.ObjectId, ref: 'Product', required: true },
      quantity: { type: Number, required: true },
    },
  ],
  status: { type: String, default: 'Pendiente' },
  createdAt: { type: Date, default: Date.now },
});

module.exports = mongoose.model('Order', OrderSchema);
